pkg/doc/signature/jsonld/processor.go

```diff
@@ -187,6 +187,36 @@ func (p *Processor) Compact(input, context map[string]interface{},
 	return proc.Compact(input, context, options)
 }
 
+// Frames makes a frame from the input view using frameDoc.
+func (p *Processor) Frame(view []string, framelDoc map[string]interface{},
+	opts ...ProcessorOpts) (map[string]interface{}, error) {
+	proc := ld.NewJsonLdProcessor()
+	options := ld.NewJsonLdOptions("")
+	options.ProcessingMode = ld.JsonLd_1_1
+	options.Format = format
+	options.ProduceGeneralizedRdf = true
+
+	procOptions := prepareOpts(opts)
+
+	useDocumentLoader(options, procOptions.documentLoader, procOptions.documentLoaderCache)
```

Comment:
@kdimak the `CachingDocumentLoader` is not thread-safe (see comments I left in #1833)

Reply:
@llorllale I absolutely agree that's no safe. But let's fix this all together for https://github.com/hyperledger/aries-framework-go/issues/1833